src/mpl.cpp

```diff
@@ -0,0 +1,264 @@
+#include "libhal-mpl/mpl.hpp"
+
+#include "mpl_reg.hpp"
+
+using namespace std::literals;
+namespace hal::mpl {
+
+mpl::mpl(hal::i2c& p_i2c)
+    : m_i2c(&p_i2c)
+{}
+
+result<mpl> mpl::create(hal::i2c& i2c)
+{
+    mpl mpl_dev(i2c);
+    if (!mpl_dev.driver_configure()) {
+        return hal::new_error();
+    }
+
+    return mpl_dev;
+}
+
+/*
+* @brief Startup configuration of MPLX device.
+    - Perform WHOAMI check
+    - Trigger reset and wait for completion
+    - Set altimeter mode
+    - Set oversampling ratio to 2^128 (OS128)
+    - Enable data ready events for pressure/altitude and temperature
+*/
+hal::status mpl::driver_configure()
+{
+    // sanity check
+    std::array<hal::byte, 1> whoami_payload { WHOAMI_R };
+    std::array<hal::byte, 1> whoami_buffer {};
+    HAL_CHECK(hal::write_then_read(*m_i2c, device_address, whoami_payload, whoami_buffer, hal::never_timeout()));
+
+    if (whoami_buffer[0] != 0xC4) {
+        return hal::new_error();
```

Comment:
```suggestion
        return hal::new_error(std::errc::no_such_device);
```

See https://en.cppreference.com/w/cpp/error/errc for a list of error codes. Emitting an empty new_error() in a driver isn't means that user cannot determine any information about the error, just that an error occurred. It would be better to at least transmit an error code. In this example we can emit `std::errc::no_such_device`. 

We haven't come up with a design language for which errors should be returned at what time, so this is still an active area of work.